### PR TITLE
UN-418: retry server >= 500 status code error

### PR DESCRIPTION
**Pull request by veqryn**
This change is [Reviewable](https://reviewable.io/reviews/contextio/contextio-go/8)

### CODE REVIEW

File: cioutil/request.go [hunk 1]
```diff
@@ -29,13 +29,26 @@ func (cio Cio) DoFormRequest(request ClientRequest, result interface{}) error {
...
-	var bodyReader io.Reader
 	bodyValues := FormValues(request.FormValues)
 	bodyString := bodyValues.Encode()
+	logRequest(cio.Log, request.Method, cioURL, bodyValues)
+
+	err := cio.createAndSendRequest(request, cioURL, bodyString, bodyValues, result)
+
+	// Retry if Status Code >= 500 and RetryServerErr is set to true
+	if cio.RetryServerErr && ErrorStatusCode(err) >= 500 {
+		err = cio.createAndSendRequest(request, cioURL, bodyString, bodyValues, result)
```
Comment (shinsato) [inline cioutil/request.go]: Do we want to add a small wait time to the retry?  Say 1 second just in case it was a network blip?

Comment (veqryn) [inline cioutil/request.go]: Sounds reasonable. 1 second, or more?

**Comment by veqryn** [comment]
Review status: 0 of 2 files reviewed at latest revision, 1 unresolved discussion.

---

*[cioutil/request.go, line 40 \[r1\]](https://reviewable.io:443/reviews/contextio/contextio-go/8#-KNAKQ-fXRmXLjVyHEDB-r1-40:-KNANOqK7RfgvtQ0y30R:-499146009) ([raw file](https://github.com/contextio/contextio-go/blob/6f1e67dae57ae1848f79ad86085fde7d0832ba11/cioutil/request.go#L40)):*
<details><summary><i>Previously, veqryn (Chris Duncan) wrote…</i></summary>
> Sounds reasonable. 1 second, or more?
</details>
Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/contextio/contextio-go/8)*

**Comment by veqryn** [comment]
Reviewed 1 of 2 files at r1, 1 of 1 files at r2.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/contextio/contextio-go/8)*

**Pull request merged**